sdcm/utils/perftune_validator.py

```diff
@@ -0,0 +1,238 @@
+import logging
+import random
+import traceback
+from functools import cached_property
+
+
+import yaml
+from deepdiff import DeepDiff
+
+from sdcm.sct_events import Severity
+from sdcm.sct_events.system import PerftuneResultEvent
+from sdcm.utils.version_utils import get_systemd_version
+
+
+# https://docs.scylladb.com/stable/operating-scylla/admin-tools/perftune.html
+
+
+PERFTUNE_LOCATION = "/opt/scylladb/scripts/perftune.py"
+TEMP_PERFTUNE_YAML_PATH = "/tmp/perftune.yaml"
+PERFTUNE_EXPECTED_RESULTS_PATH = "defaults/perftune_results.json"
+
+
+def get_machine_architecture_type(node):
+    result = node.remoter.run("uname -m")
+    return result.stdout.strip()
+
+
+def get_number_of_cpu_cores(node) -> int:
+    result = node.remoter.run("grep -c ^processor /proc/cpuinfo")
+    cores_num = int(result.stdout)
+    return cores_num
+
+
+class PerftuneExpectedResult:
+    def __init__(self, number_of_cpu_cores, nic_name, architecture):
+        self.nic_name = nic_name
+        self.total_cpu_bits = number_of_cpu_cores
+        self.architecture = architecture
+        self.expected_compute_bits: int = 0
+
+        match number_of_cpu_cores:
```

Comment:
@roydahan 

here's the new logic, it's just maps to how many irq cpus we expect.